### PR TITLE
Frontend Refactor

### PR DESCRIPTION
**Pull request by charlotte-zhuang**
1. Made `src/components/Navigation.jsx` change based on login status
2. Made logout button work
3. Refactored frontend to make them more regular

Please check that pages look okay both logged in and logged out.

### CODE REVIEW

**Review by isabellapiz — APPROVED** [review]

File: server/models/Pt.js [hunk 1]
```diff
@@ -21,7 +22,7 @@ const PtSchema = new Schema(
       type: Number,
     },
     location: {
-      type: String,
+      type: PointSchema,
```
Comment (JoshuaLawson8) [inline server/models/Pt.js:25]: what is a pointSchema?

File: src/components/Chat.jsx [hunk 2]
```diff
@@ -9,10 +9,9 @@ import React from "react";
 import {Navigation} from "./index";
 
 const ChatApp = (props) => {
-    console.log(props.user)
     return (
         <>
-        <Navigation />
+        <Navigation {...props} />
```
Comment (JoshuaLawson8) [inline src/components/Chat.jsx]: why is props spread here

File: src/components/Requests/Request.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    const FirstName = "FirstName";
+    const LastName = "LastName";
+
+    return (
+        <div className="request">
+            {/* <Navigation {...props} /> */}
+            {/* NEED TO CONDITIONALLY RENDER THESE COMPONENTS */}
+            {/* Fetch from DB: This user's list of requests, should return the names of those client users which we can render */}
+            {/* OnClick method on pt page should send the name of the client to the DB */}
+
```
Comment (JoshuaLawson8) [inline src/components/Requests/Request.jsx]: ?

File: src/components/Workout/index.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+export { default as CreateWorkoutTask } from './create-workout-task';
+export { default as EditWorkoutTask } from './edit-workout-task';
```
Comment (JoshuaLawson8) [inline src/components/Workout/index.js]: is this another export file

**Review by JoshuaLawson8 — APPROVED** [review]
A lot of good QOL

File: src/components/Requests/Request.jsx (hunk 3 above)
Comment (isabellapiz) [reply src/components/Requests/Request.jsx]: these are just comments I left myself for the requests page, it's not done yet

File: src/App.jsx [hunk 5]
```diff
@@ -1,39 +1,103 @@
...
-  return <Login user={user} setUser={x => setUser(x)} />;
+    return (
+      <Routes>
+        <Route
+          path='/'
+          element={<Home user={user} handleLogout={handleLogout} />}
+        />
+        <Route
+          path='/About'
+          element={<About user={user} handleLogout={handleLogout} />}
```
Comment (euniceyoh) [inline src/App.jsx:43]: why do we need to pass in handleLogout to each element in the nav bar?

File: src/components/Home.jsx [hunk 6]
```diff
@@ -1,28 +1,18 @@
...
-            <h1 className="font-weight-light">Home</h1>
-            <p>
-              This is the home page.
-            </p>
-          </div>
+    <>
+      <Navigation {...props} />
+      <div className='home'>
+        <div className='col-lg-5'>
+          <h1 className='font-weight-light'>Welcome!</h1>
```
Comment (euniceyoh) [inline src/components/Home.jsx]: i think we can remove this line, it's not being used bc it's blocked by the picture

**Comment by euniceyoh** [comment]
Screen Shot 2022-04-21 at 11 42 23 AM

this is what i see when i run the isabella branch. there's like a white empty box on the right

File: src/components/Home.jsx (hunk 6 above)
Comment (isabellapiz) [reply src/components/Home.jsx]: yeah I know, that line can be removed in a future merge. same with the picture not lining up, the styling isn't perfect it's just a rough start but it's fine for this merge

**Review by euniceyoh — APPROVED** [review]

File: server/models/Pt.js (hunk 1 above)
Comment (christinang3030) [reply server/models/Pt.js:25]: It’s used so that we can use geojson to provide location coordinates.

File: src/components/Workout/index.js (hunk 4 above)
Comment (charlotte-zhuang) [reply src/components/Workout/index.js]: yes, removed and consolidated with the original one.

**Review by christinang3030 — APPROVED** [review]
looks good

File: src/components/Home.jsx (hunk 6 above)
Comment (charlotte-zhuang) [reply src/components/Home.jsx]: fixed

**Pull request merged**